src/components/BackOffice/AddSORCodes/index.js

```diff
@@ -90,50 +122,172 @@ const AddSORCodes = () => {
+      formErrors.contract = 'Please select a contract'
     }
 
     if (selectedTrade === null) {
-      newErrors.trade = 'Please select a trade'
+      formErrors.trade = 'Please select a trade'
+    }
+
+    // Create array to track potential errors with new SOR codes
+    formErrors.sorCodesErrors = []
+
+    state.sorCodes.forEach((sorCode) => {
+      const sorCodeErrorsObject = checkNewSORCodeForErrors(sorCode)
+
+      // Now we check if the SorCodeErrorObject has any errors (any properties set to 'true'), if yes it's an error.
+      if (!isNewSORCodeValid(sorCodeErrorsObject)) {
+        formErrors.sorCodesErrors.push(sorCodeErrorsObject)
+      }
+    })
+
+    return formErrors
+  }
+
+  const checkNewSORCodeForErrors = (sorCode) => {
+    // We have a look at each property/field of each new SOR code.
+    // If the form field value of a property is falsy, we mark it as true,
+    // meaning there's an error with it (most likely field left blank)
+    const sorCodeErrorsObject = {
+      id: sorCode.id,
+      code: !sorCode.code ? true : false,
+      cost: !sorCode.cost ? true : false,
+      standardMinuteValue: !sorCode.standardMinuteValue ? true : false,
+      shortDescription: !sorCode.shortDescription ? true : false,
+      longDescription: !sorCode.longDescription ? true : false,
+    }
+
+    return sorCodeErrorsObject
+  }
+
+  const isNewSORCodeValid = (sorCodeErrorsObject) => {
+    // Any property set to 'true' represents an error
+    for (const property in sorCodeErrorsObject) {
+      if (property == 'id') continue
+      if (sorCodeErrorsObject[property] == true) return false
     }
 
-    return newErrors
+    return true
+  }
+
+  const validateForm = () => {
+    const formErrors = checkFormForErrors()
+    setErrors(formErrors)
+
+    // The 'Add SOR codes' button would be disabled, but if this was enabled via dev tools for any reason,
+    // this prevents a web request with no SOR codes to be sent.
+    if (state.sorCodes.length < 1) {
+      return false
+    }
+
+    // If the form is valid, formErrors will have a key of "sorCodesErrors" set to "[]".
+    // The empty array indicates no errors were found with the new added SOR codes.
+    if (
+      Object.keys(formErrors).length === 1 &&
+      Object.keys(formErrors).includes('sorCodesErrors')
+    ) {
+      if (formErrors.sorCodesErrors.length > 0) {
+        return false
+      } else {
+        return true
+      }
+    } else {
+      return false
+    }
   }
 
   const handleSubmit = async (e) => {
     e.preventDefault()
     setRequestError(null)
     setFormSuccess(null)
 
-    if (loading || fileLoading) return
+    // Used for CSV bulk upload
+    // if (loading || fileLoading) return
 
-    var errors = validate()
-    setErrors(errors)
+    const formValid = validateForm()
 
-    // there must be no errors
-    if (Object.keys(errors).length > 0) return
+    if (formValid == false) return
```

Comment:
You could simplify this with `if (!validateForm())`

Reply:
Damn I've only just seen this (post merge), but good spot